trap.go

```diff
@@ -68,13 +68,19 @@ func mkTrap(ptr *C.wasm_trap_t) *Trap {
 }
 
 func (t *Trap) ptr() *C.wasm_trap_t {
+	if t == nil {
+		return nil
+	}
```

Comment:
I don't believe this is idiomatic in Go and nil should be checked in advance, not within the method.

Reply:
Yes I'd prefer to remove these `nil` checks and fix callers since these methods should never be called with a `nil` pointer.